### ISSUE TITLE
Unable to connect to upload/connect to printer. Connection testing ok from app but will not upload.(3 rh icons stay greyed out)

### ISSUE THREAD

**Issue by stewrat2** [p3d-dev/Print-To-3D#3]
IOS 16.1. Stopped working after upgrade to v3.7

**Comment by p3d-dev** [comment]
Sorry to hear, that you have encountered an issue. On our side, we cannot see this issue, running iPadOS 16.1 and v3.7. Could you please explain, what „ 3 rh icons“ means ? Or even better send a screenshot or short video, so we can see the issue happening on your side ?

**Comment by stewrat2** [comment]
Everything up to slicing and icon creation ok. Connection and authorisation check ok. Attached photo shows ‘icons on right’ marked. Uninstalled , reinstalled with no success!RegardsStewart Windsor Sent from my iPadOn 29 Oct 2022, at 15:55, p3d-dev ***@***.***> wrote:﻿
Sorry to hear, that you have encountered an issue. On our side, we cannot see this issue running iPasOS 16.1 and v3.7. Could you please explain, what „ 3 rh icons“ means ? Or even better send a screenshot or short video, so we can see the issue happening on your side ?

**Comment by p3d-dev** [comment]
The photo is located where ? We cannot see it and the attachment is not visible on github.

**Comment by stewrat2** [comment]
Sorry. Here you go!Sent from my iPadOn 29 Oct 2022, at 17:33, p3d-dev ***@***.***> wrote:﻿
The photo is located where ? We cannot see it and the attachment is not visible on github.

**Comment by p3d-dev** [comment]
Do you use the github ipad App ? Seems the photo attachment is not properly forwarded to github. Or at least, we cannot find it.

Sorry for bothering you again for this photo.

**Comment by stewrat2** [comment]
Sent from my iPadOn 29 Oct 2022, at 17:56, p3d-dev ***@***.***> wrote:﻿
Do you use the github ipad App ? Seems the photo attachment is not properly forwarded to github. Or at least, we cannot find it.
Sorry for bothering you again for this photo.

**Comment by stewrat2** [comment]
Sent as email again. Can’t work out how to get photo int GitHub app,sorry!NSent from my iPadOn 29 Oct 2022, at 15:55, p3d-dev ***@***.***> wrote:﻿
Sorry to hear, that you have encountered an issue. On our side, we cannot see this issue running iPasOS 16.1 and v3.7. Could you please explain, what „ 3 rh icons“ means ? Or even better send a screenshot or short video, so we can see the issue happening on your side ?

**Comment by p3d-dev** [comment]
Ok. As this does not work, let us try by text.
On the right side, there are six icons, if thumbnail is enabled:
1. slicing
2. snapshot
3. info
4. slice filter
5. printer
6. share

Our understanding from your description is, that 1/2/3/4 are working as expected.
icon 5 is gray ?
If icon 5 is working, do you reach the upload view ?
icon 6 is working ?

**Comment by stewrat2** [comment]
1  & 2 are working.3,4,5 & 6 are not!Sent from my iPadOn 29 Oct 2022, at 18:36, p3d-dev ***@***.***> wrote:﻿
Ok. As this does not work, let us try by text.
On the right side, there are six icons, if thumbnail is enabled:

slicing
snapshot
info
slice filter
printer
share

Our understanding from your description is, that 1/2/3/4 are working as expected.
icon 5 is gray ?
If icon 5 is working, do you reach the upload view ?
icon 6 is working ?

**Comment by p3d-dev** [comment]
Thanks for the feedback. There is a common flag, which controls the disabled state for those buttons. Between V3.6 and V3.7 have been done few changes in regard to this flag. Why in your setup the flag is apparently not updated at all, is still a miracle to us.

Anyway, we have cleaned up the code for management of this flag, which hopefully fixes this phenomenon. As the problem does not appear on our side, we cannot verify, that this change is successful. Due to few other fixes in the pipeline, we will directly push out v3.8 to Apple for release in the AppStore.

Looking forward to your feedback for v3.8

**Comment by stewrat2** [comment]
Hi. Working again! After I upgraded to IOS16.1 I set up a printer for Tevo Tarrantula and tried to slice and upload. As I reported to you it wouldn’t connect ,even though it recognised printer on the network. This morning thought I’d try on my old iPad Pro with iOS 15.7 and it worked! I had set printer up with basic Prusa i3 template. Set up new printer on my latest iPad Pro 11” running ios16.1 ,as a Prusa i3 and tested. It worked flawlessly! Sorry for any grief I’ve caused you and thank you for your prompt reply. Best RegardsStewart Windsor Sent from my iPadOn 29 Oct 2022, at 21:31, p3d-dev ***@***.***> wrote:﻿
Thanks for the feedback. There is a common flag, which controls the disabled state for those buttons. Between V3.6 and V3.7 have been done few changes in regard to this flag. Why in your setup the flag is apparently not updated at all, is still a miracle to us.
Anyway, we have cleaned up the code for management of this flag, which hopefully fixes this phenomenon. As the problem does not appear on our side, we cannot verify, that this change is successfully. Due to few other fixes in the pipeline, we will directly push out v3.8 to Apple for release in the AppStore.
Looking forward to your feedback for v3.8
